api/src/test/java/io/druid/data/input/impl/TimestampSpecTest.java

```diff
@@ -37,6 +39,29 @@ public void testExtractTimestamp() throws Exception
   }
 
   @Test
+  public void testExtractTimestampWithTimezone() throws Exception
+  {
+    final DateTimeZone saved = DateTimeZone.getDefault();
+    final DateTimeZone dataTimezone = DateTimeZone.forID("Asia/Shanghai");
+    try {
+      DateTimeZone.setDefault(DateTimeZone.UTC);
```

Comment:
This test is designed to simulate the case when server's timezone (UTC) is different than data's timezone (Asia/Shanghai) and there's no timezone information in the data's representation.

The goal is to make sure that after we specify an explicit timezone argument, the resulting extracted timestamp represent the same instant as when the event was generated.